contracts/modules/STO/USDTiered/USDTieredSTO.sol

```diff
@@ -278,20 +297,28 @@ contract USDTieredSTO is USDTieredSTOStorage, STO {
         uint256 tempSold;
         uint256 remainingTokens;
         for (uint256 i = 0; i < tiers.length; i++) {
-            remainingTokens = tiers[i].tokenTotal.sub(tiers[i].mintedTotal);
+            remainingTokens = tiers[i].tokenTotal.sub(tiers[i].totalTokensSoldInTier);
             tempReturned = tempReturned.add(remainingTokens);
-            tempSold = tempSold.add(tiers[i].mintedTotal);
+            tempSold = tempSold.add(tiers[i].totalTokensSoldInTier);
             if (remainingTokens > 0) {
-                tiers[i].mintedTotal = tiers[i].tokenTotal;
+                tiers[i].totalTokensSoldInTier = tiers[i].tokenTotal;
             }
         }
-        address walletAddress = (treasuryWallet == address(0) ? IDataStore(getDataStore()).getAddress(TREASURY) : treasuryWallet);
+        address walletAddress = getTreasuryWallet();
         require(walletAddress != address(0), "Invalid address");
+        if (preMintAllowed)
+            tempReturned = securityToken.balanceOf(address(this));
```

Comment:
In practice `tempReturned == securityToken.balanceOf(address(this))` before this assignment right? (maybe apart from dust rounding issues)?

Reply:
I don't think so, tempReturned will be 0 before the assignment if the (preMintAllowed == true). and after that, I am tackling the rounding issues by making it granular. Or you are suggesting something else?

Reply:
Why would it be zero? I think `tiers[i].tokenTotal` and `tiers[i].totalTokensSoldInTier` are still updated during the STO if `preMintAllowed == true`?

Reply:
I guess we need this if the cap was decreased before the STO starts (in which case the balance may not match the difference between the cap(s) and the number of tokens sold)?